src/test/client.rs

```diff
@@ -845,3 +845,21 @@ async fn retry_commit_txn_check_out() {
         .await
         .expect("should see checked out event");
 }
+
+/// Verifies that `Client::shutdown` succeeds.
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn manual_shutdown() {
+    let _guard = LOCK.run_concurrently().await;
+    let client = Client::test_builder().build().await.into_client();
```

Comment:
Can we add some extra logic here to create/drop the objects which register async drop tokens on the client and make sure shutdown works as intended? And ditto below to ensure that not dropping those things won't cause a hang for `shutdown_immediate`.

Also, this might be overkill, but it could be nice to use command monitoring to make sure that the cleanup stuff actually happens when shutdown is/isn't called; I think we could just listen for a killCursors (for cursor drop) and a delete (for gridfsuploadstream drop).

I trust that everything works now, but I find this async stuff to be tricky to get right so it would be nice to have some sanity checks for any mucking around we do with these implementations in the future 🙂